### ISSUE TITLE
Property 'xx' does not exist on type 'Validation<ValidationArgs<unknown>, unknown>'

### ISSUE THREAD

**Issue by titou10titou10** [vuelidate/vuelidate#1042]
Properties defined in the`validation()`method are not recognized by typescript and so compilation fails

**Expected behavior**
Properties defined in the `validation()`methods recognized as attributes of the v$ instance

Q: How to declare the properties under vuelidate validation valid v$ attributes?

Snippet: 

    <template>
       <div v-if="v$.xx.$error">xx error</div>
    --> Property 'xx' does not exist on type 'Validation<ValidationArgs<unknown>, unknown>' <--
    </template>

    <script lang="ts">
    import useVuelidate from '@vuelidate/core'
    import { required } from '@vuelidate/validators'
    export default defineComponent({
      setup() {
        const v$ = useVuelidate()
        return { v$ }
      },
      data() {
        return {
          xx: '',
        }
      },
      validations() {
        return {
          xx: { required },
        }
      }
    }
    
Same error without the`setup()` method and exposing`v$`in`data()`:

    data() {
        return {
          v$: useVuelidate(),
        }

package.json: 

    
    {
      "dependencies": {
        "@vuelidate/core": "^2.0.0-alpha.40",
        "@vuelidate/validators": "^2.0.0-alpha.28",
        "vee-validate": "^4.5.11",
        "vue": "^3.2.32",
      },
      "devDependencies": {
        "@types/vuelidate": "^0.7.15",
        "typescript": "^4.6.3",
        "vite": "^2.9.5",
      }
    }

**Comment by dobromir-hristov** [comment]
This is very odd. @teziovsky @alexejrotar you guys have any ideas? Should he use the composition API instead to get those errors fixed?

**Comment by alexejrotar** [comment]
_please read the following as educated guessing rather than a definite answer_
my first guess is the same: composition API should fix this. I think TS isn't able to infer the type of `v$` because it is determined when calling `useValidate`. I think it is inferred from the arguments. when calling `const v$ = useValidate(data, schema)`, TS infers the type of `v$` from the type of `data`. if called with no arguments, it can't know. maybe specifying the type explicitly at creation would work, but personally I don't see how this would be better than using the composition API.

**Comment by alexejrotar** [comment]
just checked it out. composition API fixes the problem:

```
<script setup lang="ts">
import { useVuelidate } from "@vuelidate/core";
import { required } from "@vuelidate/validators";

const data = {
  foo: "",
};
const rules = {
  foo: { required },
};
const v$ = useVuelidate(data, rules);
</script>

<template>
  <div v-if="v$.foo.$error">foo is required</div>
</template>
```
I don't know TS and vue well enough to say whether/how it is possible to make TS work with the options API. I guess it's best to stick to the compositions API when working with TS

**Comment by titou10titou10** [comment]
Yes it is possible. All of our apps are with vue + options API + typescript
IMHO, all the examples on the web site  should uses TypeScript instead of plain JavaScript. Currently they all use vue + options API + plain JavaScript
I could understand that when using TypeScript vuelidate only "works" with composition API, then  in that case, the restriction  should explicitly be mentioned at the beginning of the documentation:

Vuelidate 2.x works only when:

 - vue 3 + JavaScript + options API
 - vue 3 + JavaScript + composition API
 - vue 3 + TypeScript + composition API

As the "recommended" way to use vue 3 is with TypeScript, when using vuelidate it is mandatory to use composition API even with vue, composition API is just "if you want to" but not mandatory at all

**Comment by alexejrotar** [comment]
> Yes it is possible. All of our apps are with vue + options API + typescript

could you post a snippet? I wonder how you made it work. I had the same error as in your initial post. also, how do you setup TS? I noticed once that I was able to compile vuelidate even with a type error because they weren't checked properly.

**Comment by jorparad** [comment]
I'm following the examples in the documentation using useVuelidate() inside setup() and getting this error. 

How is the correct way of specifying the type if the way described in the documentation is not working?

**Comment by alexejrotar** [comment]
@jorparad are you using options API or composition API? I suppose you are using typescript, otherwise you wouldn't get such an error. in that case I recommend using the composition API. i.e. create your data and rules inside your setup method, and call useVuelidate with these as arguments. see my previous comment for an example. if that won't solve the problem, would you post a code snippet?
